com.unity.netcode.gameobjects/Editor/CodeGen/INetworkSerializableILPP.cs

```diff
@@ -141,26 +146,56 @@ public override ILPostProcessResult Process(ICompiledAssembly compiledAssembly)
+                                            }
+                                            else if (underlyingType.HasInterface(m_INativeListBool_TypeRef.FullName) && underlyingType.HasInterface(CodeGenHelpers.IUTF8Bytes_FullName))
+                                            {
+                                                if (underlyingType.IsGenericInstance)
+                                                {
+                                                    fixedStringTypesSet.Add(underlyingType);
+                                                }
+                                            }
 
-                                                if (underlyingType.Resolve().IsEnum)
+                                            else if (underlyingType.Resolve().IsEnum)
+                                            {
+                                                if (underlyingType.IsGenericInstance)
                                                 {
                                                     enumTypesSet.Add(underlyingType);
                                                 }
                                             }
+                                            else if (!underlyingType.Resolve().IsPrimitive)
+                                            {
+                                                bool methodExists = false;
+                                                foreach (var method in m_FastBufferWriterType.Methods)
+                                                {
+                                                    if (!method.HasGenericParameters && method.Parameters.Count == 1 && method.Parameters[0].ParameterType.Resolve() == underlyingType.Resolve())
+                                                    {
+                                                        methodExists = true;
+                                                        break;
+                                                    }
+                                                }
+
+                                                if (!methodExists)
+                                                {
+                                                    m_Diagnostics.AddError($"{type}.{field.Name}: {underlyingType} is not valid for use in {typeof(NetworkVariable<>).Name} types. Types must either implement {nameof(INetworkSerializeByMemcpy)} or {nameof(INetworkSerializable)}. If this type is external and you are sure its memory layout makes it serializable by memcpy, you can replace {underlyingType} with {typeof(ForceNetworkSerializeByMemcpy<>).Name}<{underlyingType}>, or you can create extension methods for {nameof(FastBufferReader)}.{nameof(FastBufferReader.ReadValueSafe)}(this {nameof(FastBufferReader)}, out {underlyingType}) and {nameof(FastBufferWriter)}.{nameof(FastBufferWriter.WriteValueSafe)}(this {nameof(FastBufferWriter)}, in {underlyingType}) to define serialization for this type.");
```

Comment:
Nit, but have you thought about having a short error message prefixing the detailed one bhere so that it's to refer to in community messaging?
"Unsupported Network Serialization Type. {type}.{field.Name}: {underlyingType} is not..."

Reply:
I've got a follow-up PR that gets rid of this message again and moves the info elsewhere, so let's revisit the wording in that PR if it's still not good for you.